### PR TITLE
chore: In gnodev, include gnoweb flags

### PR DESCRIPTION
**Pull request by jefft0**
gnodev starts gnoweb which can show a [help page](https://gno.berty.io/r/berty/social?help) with the gnokey command for each function. The help for the gnokey command includes `-remote` which is the same as the gnodev listening address. But the gnokey command is meant to be used from a different computer, and the address which gnodev listens on can be different than the address to connect from outside. (In our use case, gnodev runs in a docker container behind a reverse proxy.)

The gnoweb command already has an option `-help-remote` which is used on the help page. When gnodev [starts gnoweb](https://github.com/gnolang/gno/blob/ccc6d5b789bbb6a267135b05047db8b85adc74c7/contribs/gnodev/cmd/gnodev/setup_web.go#L15), it gives its own listening address. This PR adds a gnodev command line option `-web-help-remote` . If supplied, this overrides and is used when starting gnoweb.

### CODE REVIEW

**Comment by gfanton** [comment]
Thinking more about this, I'm not really sure about adding this. I don't want to have to duplicate every flag of `gnoweb` into `gnodev`. What do you think about simply running a separate instance of `gnoweb` on the side targeting your `gnodev` instance?

**Comment by jefft0** [comment]
@gfanton, I don't see an option in gnodev to _not_ start gnoweb. What should we do if we run gnoweb separately?

**Comment by jefft0** [comment]
Thanks. Sounds good. Do you want me to change this PR to "Add an option to disable `gnoweb` in gnodev" ? 
What should the gnodev option be? There is already an option `-web-listener` . Should we allow `-web-listener none` ?

**Comment by D4ryl00** [comment]
If I can argue, I think `gnodev` should handle every service for us, like a hub. So it would stay simple for us to deploy the services through gnodev.
I don't see what would be the problem to recreate each option in `gnodev` for each services. So for us (devs), we would have a central configuration for all services. It would be great. If you categorize the options list in `gnodev --help`, it would be easy to find them. No matter if there are 100 options. The more there are, the more gnodev would be useful.

It's my opinion.

**Comment by moul** [comment]
Gnodev should be the go-to tool for developers to work locally and showcase their projects. Running multiple manual containers should be considered advanced, while using gnodev in Docker can be a standard method to isolate multiple services on a server. Therefore, I recommend enabling gnodev to configure gnoweb accordingly.

People developing or running standalone Gnoweb should be experienced individuals or actual Gnoweb developers, in my opinion.

File: contribs/gnodev/cmd/gnodev/main.go [hunk 1]
```diff
@@ -51,13 +51,14 @@ type devCfg struct {
...
-	chainId    string
-	serverMode bool
+	minimal       bool
+	verbose       bool
+	noWatch       bool
+	noReplay      bool
+	maxGas        int64
+	chainId       string
+	serverMode    bool
+	webHelpRemote string
```
Comment (moul) [inline contribs/gnodev/cmd/gnodev/main.go:61]: Embed `gnoweb.Config` instead of using individual gnoweb flags.

Comment (moul) [reply contribs/gnodev/cmd/gnodev/main.go:61]: `gnodev.Config` should include specific flags and embed configurations like `gnoweb.Config`, `gnoland.Config`, etc.

File: contribs/gnodev/cmd/gnodev/setup_web.go [hunk 2]
```diff
@@ -12,7 +12,11 @@ import (
 func setupGnoWebServer(logger *slog.Logger, cfg *devCfg, dnode *gnodev.Node) http.Handler {
 	webConfig := gnoweb.NewDefaultConfig()
 	webConfig.RemoteAddr = dnode.GetRemoteAddress()
-	webConfig.HelpRemote = dnode.GetRemoteAddress()
+	if cfg.webHelpRemote != "" {
+		webConfig.HelpRemote = cfg.webHelpRemote
+	} else {
+		webConfig.HelpRemote = dnode.GetRemoteAddress()
+	}
```
Comment (moul) [inline contribs/gnodev/cmd/gnodev/setup_web.go:19]: Initialize `webConfig` on the main struct and manage all flag updates using flags. In essence, inherit and share flag logic from subcomponents like gnoweb, gnoland, etc.

File: docs/gno-tooling/cli/gnodev.md [hunk 3]
```diff
@@ -121,3 +121,4 @@ While `gnodev` is running, the following shortcuts are available:
 | --server-mode       | disable interaction, and adjust logging for server use.    |
 | --verbose           | enable verbose output for development                      |
 | --web-listener      | web server listening address                               |
+| --web-help-remote   | web server help page's remote addr                         |
```
Comment (moul) [inline docs/gno-tooling/cli/gnodev.md:124]: Eventually, we can use `--web-` or `--gnoweb-` as prefixes for Gnoweb, and `--node-` or `--gnoland-` for the node.

File: contribs/gnodev/cmd/gnodev/main.go (hunk 1 above)
Comment (jefft0) [reply contribs/gnodev/cmd/gnodev/main.go]: Added a commit to embed gnoweb.Config

File: contribs/gnodev/cmd/gnodev/setup_web.go (hunk 2 above)
Comment (jefft0) [reply contribs/gnodev/cmd/gnodev/setup_web.go]: Now initializing in main `webConfig: gnoweb.NewDefaultConfig(),`

File: docs/gno-tooling/cli/gnodev.md (hunk 3 above)
Comment (jefft0) [reply docs/gno-tooling/cli/gnodev.md:139]: Agree. For now, `web-help-remote` is the only gnoweb option that makes sense for gnodev. Later we can add others like `web-captcha-site` if needed.

**Comment by gfanton** [comment]
@moul I've discussed with @jefft0 and I agree we can include those flags in gnodev as it doesn't cost much yet. I just want to avoid duplicating a lot of documentation/code and reaching a point where we have to maintain it symmetrically.

**Review by gfanton — APPROVED** [review]

**Pull request merged**